core/chains/evm/client/client.go

```diff
@@ -27,7 +27,7 @@ const queryTimeout = 10 * time.Second
 
 // Client is the interface used to interact with an ethereum node.
 type Client interface {
```

Comment:
Add this too:
var _ cmntypes.Client = (*client)(nil)

Reply:
https://github.com/smartcontractkit/chainlink/blob/363ac54098a9a96f3806b31a7bc347944ab55faf/core/chains/evm/client/client.go#L91

Reply:
That one is evm client interface.
You also want to add the new client interface that you just created. So both of them.

Reply:
Hmm, I don't think I follow. Generic client interface is an inherit part of this evm client, so this check validate both. Or did you mean something else?

Reply:
Ohh, so the EVM client implements the Generic client and not the evm struct.
Ya sorry, my mistake.